packages/payload/src/utilities/flattenTopLevelFields.ts

```diff
@@ -10,45 +14,117 @@ import type {
   Tab,
 } from '../fields/config/types.js'
 
-import {
-  fieldAffectsData,
-  fieldHasSubFields,
-  fieldIsPresentationalOnly,
-  tabHasName,
-} from '../fields/config/types.js'
+import { fieldAffectsData, fieldIsPresentationalOnly, tabHasName } from '../fields/config/types.js'
 
 type FlattenedField<TField> = TField extends ClientField
-  ? FieldAffectingDataClient | FieldPresentationalOnlyClient
-  : FieldAffectingData | FieldPresentationalOnly
+  ? { accessor?: string; labelWithPrefix?: string } & (
+      | FieldAffectingDataClient
+      | FieldPresentationalOnlyClient
+    )
+  : { accessor?: string; labelWithPrefix?: string } & (FieldAffectingData | FieldPresentationalOnly)
 
 type TabType<TField> = TField extends ClientField ? ClientTab : Tab
 
+type FlattenFieldsArgs<TField extends ClientField | Field> = {
+  fields: TField[]
+  i18n?: I18nClient
+  keepPresentationalFields?: boolean
+  labelPrefix?: string
+  pathPrefix?: string
+}
+
 /**
  * Flattens a collection's fields into a single array of fields, as long
  * as the fields do not affect data.
  *
  * @param fields
  * @param keepPresentationalFields if true, will skip flattening fields that are presentational only
  */
-function flattenFields<TField extends ClientField | Field>(
-  fields: TField[],
-  keepPresentationalFields?: boolean,
-): FlattenedField<TField>[] {
+function flattenFields<TField extends ClientField | Field>({
```

Comment:
Modifying args here would be considered a breaking change. This function exported from `payload` and used extensively throughout the Payload universe. Modifying the behavior of this function, too, should be done with great caution.